internal/cache/file/cache_handler_test.go

```diff
@@ -0,0 +1,438 @@
+}
+
+func (chrT *cacheHandlerTest) addTestFileInfoEntryInCache(bucketName string, objectName string) string {
+	// Add an entry into
+	fileInfoKey := data.FileInfoKey{
+		BucketName: bucketName,
+		ObjectName: objectName,
+	}
+	fileInfo := data.FileInfo{
+		Key:              fileInfoKey,
+		ObjectGeneration: chrT.object.Generation,
+		FileSize:         chrT.object.Size,
+		Offset:           0,
+	}
+
+	fileInfoKeyName, err := fileInfoKey.Key()
+	AssertEq(nil, err)
+
+	_, err = chrT.cache.Insert(fileInfoKeyName, fileInfo)
+	AssertEq(nil, err)
+
+	return fileInfoKeyName
+}
+
+func (chrT *cacheHandlerTest) getCacheHandleForSetupEntryInCache() *CacheHandle {
+	cacheHandle, err := chrT.cacheHandler.GetCacheHandle(chrT.object, chrT.bucket, 0)
+	AssertEq(nil, err)
+	AssertNe(nil, cacheHandle)
+	AssertEq(true, isFileExist(chrT.downloadPath))
+
+	return cacheHandle
+}
+
+func (chrT *cacheHandlerTest) getMinObject(objName string, objContent string) *gcs.MinObject {
+	ctx := context.Background()
+	objects := map[string][]byte{objName: []byte(objContent)}
+	err := storageutil.CreateObjects(ctx, chrT.bucket, objects)
+	AssertEq(nil, err)
+
+	gcsObj, err := chrT.bucket.StatObject(ctx, &gcs.StatObjectRequest{Name: objName,
+		ForceFetchFromGcs: true})
+	AssertEq(nil, err)
+	minObject := storageutil.ConvertObjToMinObject(gcsObj)
+	return &minObject
+}
+
+// isFileExist returns true if the file exists and false otherwise.
+// If an error occurs, the function panics.
+func isFileExist(filePath string) bool {
+	_, err := os.Stat(filePath)
+
+	if err == nil {
+		return true
+	}
+
+	if os.IsNotExist(err) {
+		return false
+	}
+
+	AssertEq(nil, err)
+	return false
+}
+
+func (chrT *cacheHandlerTest) Test_createLocalFileReadHandle_OnlyForRead() {
+	readFileHandle, err := chrT.cacheHandler.createLocalFileReadHandle(chrT.object.Name, chrT.bucket.Name())
+
+	ExpectEq(nil, err)
+	_, err = readFileHandle.Write([]byte("test"))
+	ExpectNe(nil, err)
+	ExpectTrue(strings.Contains(err.Error(), "bad file descriptor"))
+}
+
+func (chrT *cacheHandlerTest) Test_performPostEvictionWork() {
+	// Existing cacheHandle.
+	cacheHandle := chrT.getCacheHandleForSetupEntryInCache()
+	AssertEq(nil, cacheHandle.validateCacheHandle())
+	fileInfo := chrT.cache.LookUp(chrT.fileInfoKeyName)
+	fileInfoData := fileInfo.(data.FileInfo)
+	jobStatusBefore := cacheHandle.fileDownloadJob.GetStatus()
+	AssertEq(jobStatusBefore.Name, downloader.NOT_STARTED)
+	jobStatusBefore, err := cacheHandle.fileDownloadJob.Download(context.Background(), int64(util.MiB), false)
+	AssertEq(nil, err)
+	AssertEq(jobStatusBefore.Name, downloader.DOWNLOADING)
+
+	err = chrT.cacheHandler.performPostEvictionWork(&fileInfoData)
+
+	ExpectEq(nil, err)
+	jobStatusAfter := cacheHandle.fileDownloadJob.GetStatus()
+	ExpectEq(jobStatusAfter.Name, downloader.INVALID)
+	ExpectEq(false, isFileExist(chrT.downloadPath))
+}
+
+func (chrT *cacheHandlerTest) Test_performPostEvictionWork_WhenLocalFileNotExist() {
```

Comment:
should we not throw error in performPostEvictionWork if file is already deleted ? our intention is to delete the file and if it is already deleted, we should be good.

Reply:
Not throwing error for `file not exist`, but logged this error as warning to catch it in our testing.